src/Form/LinkBlockFormDataProvider.php

```diff
@@ -0,0 +1,272 @@
+                $customContent
+            );
+        } else {
+            $linkBlockId = $linkBlock['id_link_block'];
+            $this->repository->update(
+                $linkBlockId,
+                $linkBlock['block_name'],
+                $linkBlock['id_hook'],
+                $linkBlock['cms'],
+                $linkBlock['static'],
+                $linkBlock['product'],
+                $customContent
+            );
+        }
+        $this->setIdLinkBlock($linkBlockId);
+        $this->updateHook($linkBlock['id_hook']);
+        $this->cache->clearModuleCache();
+
+        return [];
+    }
+
+    /**
+     * @return int
+     */
+    public function getIdLinkBlock()
+    {
+        return $this->idLinkBlock;
+    }
+
+    /**
+     * @param int $idLinkBlock
+     * @return LinkBlockFormDataProvider
+     */
+    public function setIdLinkBlock($idLinkBlock)
+    {
+        $this->idLinkBlock = $idLinkBlock;
+
+        return $this;
+    }
+
+    /**
+     * @param array $data
+     * @return array
+     */
+    private function validateLinkBlock(array $data)
+    {
+        $errors = [];
+        if (!isset($data['id_hook'])) {
+            $errors[] = [
+                'key' => "Missing id_hook",
+                'domain' => 'Admin.Catalog.Notification',
+                'parameters' => [],
+            ];
+        }
+
+        if (!isset($data['block_name'])) {
+            $errors[] = [
+                'key' => "Missing block_name",
+                'domain' => 'Admin.Catalog.Notification',
+                'parameters' => [],
+            ];
+        } else {
+            foreach ($this->languages as $language) {
+                if (
+                    !isset($data['block_name'][$language['id_lang']]) ||
+                    empty($data['block_name'][$language['id_lang']])
+                ) {
+                    $errors[] = [
+                        'key' => "Missing block_name value for language %s",
+                        'domain' => 'Admin.Catalog.Notification',
+                        'parameters' => [$language['iso_code']],
+                    ];
+                }
+            }
+        }
+
+        if (isset($data['custom'])) {
+            foreach ($data['custom'] as $customIndex => $custom) {
+                foreach ($this->languages as $language) {
+                    if (!isset($custom[$language['id_lang']])) {
+                        $errors[] = [
+                            'key' => "Missing block_name value for language %s",
+                            'domain' => 'Admin.Catalog.Notification',
+                            'parameters' => [$language['iso_code']],
+                        ];
+                    } else {
+                        $langCustom = $custom[$language['id_lang']];
+                        $fields = ['title', 'url'];
+                        foreach ($fields as $field) {
+                            if (!isset($langCustom[$field]) || empty($langCustom[$field])) {
```

Comment:
!isset is unneeded because of empty test

Reply:
I'll sleep smarter tonight 😄